tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AWS/Kinesis/ContextPropagation.cs

```diff
@@ -0,0 +1,104 @@
+// <copyright file="ContextPropagation.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Datadog.Trace.DuckTyping;
+using Datadog.Trace.Logging;
+using Datadog.Trace.Propagators;
+using Datadog.Trace.Vendors.Newtonsoft.Json;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.AWS.Kinesis
+{
+    internal static class ContextPropagation
+    {
+        private const string KinesisKey = "_datadog";
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(ContextPropagation));
+
+        public static void InjectTraceIntoRecords<TRecordsRequest>(IPutRecordsRequest request, SpanContext context)
+        {
+            // request.Records is not null and has at least one element
+            if (request.Records is not { Count: > 0 })
+            {
+                return;
+            }
+
+            var record = request.Records[0].DuckCast<IContainsData>();
+            InjectTraceIntoData<TRecordsRequest>(record, context);
+        }
+
+        public static void InjectTraceIntoData<TRecordRequest>(IContainsData record, SpanContext context)
+        {
+            if (record.Data is null)
+            {
+                return;
+            }
+
+            Inject<TRecordRequest>(record, context);
```

Comment:
Since the type you're using is the type of the record object, I recommend omitting the generic type and letting the compiler infer it.

```suggestion
            Inject(record, context);
```